src/main/java/com/example/todolist/Infrastructure/Repository/TaskRepository.java

```diff
@@ -1,13 +1,20 @@
 package com.example.todolist.Infrastructure.Repository;
 
 import com.example.todolist.Domain.Entity.Task;
+import io.micrometer.common.lang.NonNull;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.PagingAndSortingRepository;
 
 public interface TaskRepository extends PagingAndSortingRepository<Task, String> {
     void save(Task task);
     Task findById(String id);
 
+    @Override
+    @NonNull
```

Comment:
what is the purpose of @NonNull here?

Reply:
It relies on the warning that: "Not annotated method overrides method annotated with @NonNullApi".